src/components/Select/Select.jsx

```diff
@@ -252,13 +268,28 @@ export default class Select extends React.Component {
     }
   };
 
+  findItemInState = item => {
+    const { selectedItems } = this.state;
+    return selectedItems.find(x => this.getItemId(x) === this.getItemId(item));
+  };
+
   onSearchChange = searchValue => {
     const { items, keyMap } = this.props;
     const searchFiled = keyMap ? keyMap.title : 'title';
 
-    const filteredItems = items.filter(item =>
-      includes(item[searchFiled].toLowerCase(), searchValue.toLowerCase())
-    );
+    const filteredItems = items.reduce((filtered, item) => {
+      if (
+        includes(item[searchFiled].toLowerCase(), searchValue.toLowerCase())
+      ) {
+        filtered.push({
+          ...item,
+          selected:
+            this.findItemInState(item) && this.findItemInState(item).selected,
+        });
+      }
```

Comment:
For each item in `props.items` , if the item contains the search string, add that item to the `filtered` list and set the `selected` state of that item to be the one from the `selectedItems`